src/AWS/Lambda/Events/ApiGateway/ProxyResponse.hs

```diff
@@ -0,0 +1,132 @@
+                                            statusIsInformational,
+                                            statusIsRedirection,
+                                            statusIsServerError,
+                                            statusIsSuccessful)
+
+-- | Type that represents the body returned to an API Gateway when using HTTP
+-- Lambda Proxy integration.  It is highly recommended that you do not use this
+-- type directly, and instead use the smart constructors exposed such as
+-- 'textPlain', 'applicationJSON', and 'genericBinary'.  These make sure that
+-- the base64 encodings work transparently.
+data ProxyBody = ProxyBody
+    { contentType     :: T.Text
+    , serialized      :: T.Text
+    , isBase64Encoded :: Bool
+    } deriving (Show)
+
+-- | A response returned to an API Gateway when using the HTTP Lambda Proxy
+-- integration.  ContentType will be set based on the ProxyBody (recommended)
+-- if a value is not present in the headers field.
+--
+-- @
+--     {-\# LANGUAGE NamedFieldPuns \#-}
+--     {-\# LANGUAGE DuplicateRecordFields \#-}
+--
+--     module Main where
+--
+--     import AWS.Lambda.Runtime (pureRuntime)
+--     import AWS.Lambda.Events.ApiGateway.ProxyRequest (ProxyRequest(..), NoAuthorizer)
+--     import AWS.Lambda.Events.ApiGateway.ProxyResponse (ProxyResponse(..), textPlain, forbidden403, ok200)
+--
+--     myHandler :: ProxyRequest NoAuthorizer -> ProxyResponse
+--     myHandler ProxyRequest { httpMethod = \"GET\", path = "/say_hello" } =
+--         ProxyResponse
+--         {   status = ok200
+--         ,   body = textPlain \"Hello\"
+--         ,   headers = mempty
+--         ,   multiValueHeaders = mempty
+--         }
+--     myHandler _ =
+--         ProxyResponse
+--         {   status = forbidden403
+--         ,   body = textPlain \"Forbidden\"
+--         ,   headers = mempty
+--         ,   multiValueHeaders = mempty
+--         }
+--
+--     main :: IO ()
+--     main = pureRuntime myHandler
+-- @
+data ProxyResponse = ProxyResponse
+    { status            :: Status
+    , headers           :: HashMap (CI T.Text) T.Text
+    , multiValueHeaders :: HashMap (CI T.Text) [T.Text]
+    , body              :: ProxyBody
+    } deriving (Show)
+
+-- | Smart constructor for creating a ProxyBody with an arbitrary ByteString of
+-- the chosen content type.
+genericBinary :: T.Text -> ByteString -> ProxyBody
+genericBinary contentType x =
+    ProxyBody contentType (TE.decodeUtf8 $ B64.encode x) True
+
+-- | Smart constructor for creating a simple body of text.
+textPlain :: T.Text -> ProxyBody
+textPlain x = ProxyBody "text/plain; charset=utf-8" x False
+
+-- | Smart constructor for creating a simple body of JSON.
+applicationJson :: ToJSON a => a -> ProxyBody
+applicationJson x =
+    ProxyBody
+        "application/json; charset=utf-8"
+        (TL.toStrict $ TLE.decodeUtf8 $ encode x)
+        False
+
+-- | Smart constructor for creating a simple body of a GIF (that has already
```

Comment:
Did we ever come down on deciding whether we should provide all these or not? Seems odd to only provide gif/jpeg at the outset hah

Reply:
Nothing firm, nope.  I wanted to have it be clear how people might use these and create their own, but we simultaneously don’t want to support every MIME type ever.

Reply:
Could we do so in the module docs instead, rather than providing examples as code that might we might change/remove and force a major version revision?